src/lib/encoder.test.ts

```diff
@@ -666,13 +667,15 @@ describe('encoder', () => {
         );
       });
 
-      it('throws when trying to decode a bytes17 as `uint128`', () => {
-        expect(() =>
-          decodeValueType('uint128', '0x000000000000000000000000000000ffff'),
-        ).to.throw(
-          "Can't convert hex value 0x000000000000000000000000000000ffff to uint128. Too many bytes. 17 > 16",
-        );
-      });
+      // We do not want this during decode. During encode we enforce the correct sizes.
+      //
+      // it('throws when trying to decode a bytes17 as `uint128`', () => {
```

Comment:
maybe we can adjust this test so that it no longer throws

Reply:
Ah good call.